File: HelperJobby/DTOs/Job/CurrentJobCreationDTO.cs
using ApplicationDomain.Enums;
using HelperJobby.DTOs.Account;

namespace HelperJobby.DTOs.Job;

public class CurrentJobCreationDTO
{
    public string JobTitle { get; set; }
    public int NumberOfOpenings { get; set; }
    public string Language { get; set; }
    public string Location { get; set; }
    public List<JobTypes> JobType { get; set; }
    public decimal Salary { get; set; }
    public List<Schedules> Schedule { get; set; }
    public List<EmployeeBenefits> Benefits { get; set; }
    public string ContactEmail { get; set; }
    public bool ResumeRequired { get; set; }
    public string Description { get; set; }
    public int EmployerAccountId { get; set; }
    public EmployerAccountDTO EmployerAccount { get; set; }
}

File: HelperJobby/client-app/src/DTOs/jobRelatetedDTOs/CurrentJobCreationDTO.ts
import {EmployerAccountDTO} from "../accountDTOs/EmployerAccountDTO";
import EmployeeBenefits from "../../enums/EmployeeBenefits";
import Schedules from "../../enums/Schedules";
import JobTypes from "../../enums/JobTypes";

export interface CurrentJobCreationDTO {
    id: number;
    jobTitle: string;
    numberOfOpenings: number;
    language: string;
    location: string;
    jobType: JobTypes[];
    salary: number;
    salaryRate: string;
    showPayBy: string;
    schedule: Schedules[];
    benefits: EmployeeBenefits[];
    contactEmail: string;
    resumeRequired: boolean;
    description: string;
    employerAccountId: number;
    employerAccount: EmployerAccountDTO;
}